internals/auth/signature_test.go

```diff
@@ -154,6 +143,92 @@ func TestVerify(t *testing.T) {
 	}
 }
 
+var testResult = &auth.Result{
+	AccountID:   uuid.New(),
+	Fingerprint: "test",
+}
+
+type fakeMethod struct{}
+
+func (m fakeMethod) Verify(credentials string, data []byte) (*auth.Result, error) {
+	return testResult, nil
+}
+
+func (m fakeMethod) Tag() string {
+	return "TestAuth"
+}
+
+func TestAuthenticator_Verify(t *testing.T) {
+	cases := map[string]struct {
+		headers  map[string]string
+		expected *auth.Result
+		err      error
+	}{
+		"outdated v1": {
+			headers: map[string]string{
+				"Authorization": "SecretHub foo:bar:baz",
+			},
+			err: auth.ErrOutdatedSignatureProtocol,
+		},
+		"outdated v2": {
+			headers: map[string]string{
+				"Authorization": "SecretHub-Sig2 foo:bar",
+			},
+			err: auth.ErrOutdatedSignatureProtocol,
+		},
+		"unsupported auth method": {
+			headers: map[string]string{
+				"Authorization": "Basic username:password",
+			},
+			err: auth.ErrUnsupportedAuthFormat,
+		},
+		"no header": {
+			err: auth.ErrNoAuthHeader,
+		},
+		"empty header": {
+			headers: map[string]string{
+				"Authorization": "",
+				"Date":          time.Now().Format(time.RFC3339),
+			},
+			err: auth.ErrNoAuthHeader,
+		},
+		"no key or token": {
+			headers: map[string]string{
+				"Authorization": "TestAuth",
+				"Date":          time.Now().Format(time.RFC3339),
+			},
+			err: auth.ErrUnsupportedAuthFormat,
+		},
```

Comment:
Should we add a check for an expired signature?

Reply:
yes

Reply:
done